internal/testrunner/runners/system/tester.go

```diff
@@ -556,18 +561,18 @@ func (r *tester) tearDownTest(ctx context.Context) error {
 		r.removeAgentHandler = nil
 	}
 
-	if r.deleteTestPolicyHandler != nil {
-		if err := r.deleteTestPolicyHandler(cleanupCtx); err != nil {
+	if r.shutdownAgentHandler != nil {
+		if err := r.shutdownAgentHandler(cleanupCtx); err != nil {
 			return err
 		}
-		r.deleteTestPolicyHandler = nil
+		r.shutdownAgentHandler = nil
 	}
 
-	if r.shutdownAgentHandler != nil {
-		if err := r.shutdownAgentHandler(cleanupCtx); err != nil {
+	if r.deleteTestPolicyHandler != nil {
```

Comment:
Reordering this because the agent deployer needs to get the YAML for the agent, and it contains an enrollment token now, that needs to be retrieved for an existing policy.

We could refactor tear down to don't require this token there, but in any case it makes sense to delete the policy after deleting the agents using it.